src/game/logic/object/locomotor.cpp

```diff
@@ -0,0 +1,812 @@
+    m_maxSpeedDamaged(-1.0f),
+    m_maxTurnRateDamaged(-1.0f),
+    m_accelerationDamaged(-1.0f),
+    m_liftDamaged(-1.0f),
+    m_surfaces(0),
+    m_maxSpeed(0.0f),
+    m_maxTurnRate(0.0f),
+    m_acceleration(0.0f),
+    m_lift(0.0f),
+    m_braking(99999.0f),
+    m_minSpeed(0.0f),
+    m_minTurnSpeed(99999.0f),
+    m_behaviorZ(Z_NO_Z_MOTIVE_FORCE),
+    m_appearance(LOCO_OTHER),
+    m_groupMovementPriority(PRIORITY_MOVES_MIDDLE),
+    m_preferredHeight(0.0f),
+    m_preferredHeightDamping(1.0f),
+    m_circlingRadius(0.0f),
+    m_maxThrustAngle(0.0f),
+    m_speedLimitZ(999999.0f),
+    m_extra2DFriction(0.0f),
+    m_accelPitchLimit(0.0f),
+    m_deaccelPitchLimit(0.0f),
+    m_bounceKick(0.0f),
+    m_pitchStiffness(0.1f),
+    m_rollStiffness(0.1f),
+    m_pitchDamping(0.9f),
+    m_rollDamping(0.9f),
+    m_forwardVelCoef(0.0f),
+    m_pitchInDirectionOfZVelFactor(0.0f),
+    m_thrustRoll(0.0f),
+    m_thrustWobbleRate(0.0f),
+    m_thrustMinWobble(0.0f),
+    m_thrustMaxWobble(0.0f),
+    m_lateralVelCoef(0.0f),
+    m_forwardAccelCoef(0.0f),
+    m_lateralAccelCoef(0.0f),
+    m_uniformAxialDamping(1.0f),
+    m_turnPivotOffset(0.0f),
+    m_apply2DFrictionWhenAirborne(false),
+    m_downhillOnly(false),
+    m_allowMotiveForceWhileAirborne(false),
+    m_locomotorWorksWhenDead(false),
+    m_airborneTargetingHeight(INT_MAX),
+    m_stickToGround(false),
+    m_canMoveBackwards(false),
+    m_hasSuspension(false),
+    m_wheelTurnAngle(0.0f),
+    m_maximumWheelExtension(0.0f),
+    m_maximumWheelCompression(0.0f),
+    m_closeEnoughDist(1.0f),
+    m_closeEnoughDist3D(false),
+    m_slideIntoPlaceTime(0.0f),
+    m_wanderWidthFactor(0.0f),
+    m_wanderLengthFactor(1.0f),
+    m_wanderAboutPointRadius(0.0f),
+    m_rudderCorrectionDegree(0.0f),
+    m_rudderCorrectionRate(0.0f),
+    m_elevatorCorrectionDegree(0.0f),
+    m_elevatorCorrectionRate(0.0f)
+{
+}
+
+void LocomotorTemplate::Validate()
+{
+    if (m_maxSpeedDamaged < 0.0f) {
+        m_maxSpeedDamaged = m_maxSpeed;
+    }
+    if (m_maxTurnRateDamaged < 0.0f) {
+        m_maxTurnRateDamaged = m_maxTurnRate;
+    }
+    if (m_accelerationDamaged < 0.0f) {
+        m_accelerationDamaged = m_acceleration;
+    }
+    if (m_liftDamaged < 0.0f) {
+        m_liftDamaged = m_lift;
+    }
+
+    switch (m_appearance) {
+        case LOCO_WINGS:
+            if (m_minSpeed <= 0.0f) {
+                captainslog_dbgassert(false, "WINGS should always have positive minSpeeds (otherwise they hover)");
+                m_minSpeed = 0.01f;
+            }
+
+            if (m_minTurnSpeed <= 0.0f) {
+                captainslog_dbgassert(false, "WINGS should always have positive minTurnSpeeds");
+                m_minTurnSpeed = 0.01f;
+            }
+
+            break;
+        case LOCO_THRUST:
+            captainslog_relassert(
+                m_behaviorZ != LocomotorBehaviorZ::Z_NO_Z_MOTIVE_FORCE || m_lift == 0.0f || m_liftDamaged == 0.0f,
```

Comment:
the case is`if ( this->m_behaviorZ || this->m_maxLift != 0.0 || this->m_maxLiftDamaged != 0.0 )` so what you have there isn't the inverse of that

Reply:
Fixed.